### DISCUSSION TITLE
make useContractWrite to wait for Metamask successful transaction confirmation

### DISCUSSION THREAD

**Discussion by 0xPro** [wevm/wagmi#3140] [Q&A]
I'm following Wagmi's [contract write example ](https://wagmi.sh/examples/contract-write) with Hardhat and am seeing that the transaction is successfully returned while Metamask is still pending. 
How can I make it so that the UI reflects Metamask status?

**Comment by Sotatek-HiepNguyen2** [comment]
You can use https://wagmi.sh/core/actions/waitForTransaction

**Reply by hammeiam** [reply]
Additionally you'll want to set `confirmations` to whatever number MetaMask uses https://wagmi.sh/core/actions/waitForTransaction#confirmations-optional

**Reply by 0xPro** [reply]
@Sotatek-HiepNguyen2 does waitForTransaction work differently from useWaitForTransaction?

@hammeiam I thought it was, but if my Hardhat is set to generate block only on transactions, there wont be any increments in block after the first successful transaction (meaning Metamask should be stuck in pending if it needs more confirmation)

**Reply by Sotatek-HiepNguyen2** [reply]
@0xPro I think it's the way `useWaitForTransaction` works under the hood :)

**Reply by 0xPro** [reply]
if that's the case it's already in the code, so I'm stumped why it is not in sync
